api/ruby/find-inactive-members/find_inactive_members.rb

```diff
@@ -0,0 +1,110 @@
+require "octokit"
+require "csv"
+
+if ARGV.length > 3 || ARGV.length == 0
+  puts "usage: ruby find_inactive_members.rb orgName YYYY-MM-DD purge(optional)"
+  exit(1)
+end
+
+if ARGV[2] == "purge"
+  print "Do you really want to purge all members of #{ARGV[0]} inactive since #{ARGV[1]}? y/n: "
+  response = STDIN.gets.chomp
+  if response != "y"
+    exit(1)
+  end
+end
+
+# initialize octokit
+Octokit.auto_paginate = true
+@client = Octokit::Client.new
+
+# get all organization members and place into an array of hashes
+@members = []
+@client.organization_members(ARGV[0]).each do |member|
+  hsh = {}
+  hsh["login"] = member["login"]
+  hsh["active"] = false
+  @members << hsh
+end
+
+# get all repos in the organizaton and place into a hash
+@repos = []
+@client.organization_repositories(ARGV[0]).each do |repo|
+  hsh = {}
+  hsh["full_name"] = repo["full_name"]
+  @repos << hsh
+end
+
+@total_repos = @repos.length
+@total_members = @members.length
+
+# print update to terminal
+puts "\n"
+puts "Analying activity for #{@total_members} members and #{@total_repos} repos in #{ARGV[0]}"
+
+@repos_completed = 0
+
+# method to switch member status to active
+def make_active(login)
+  hsh = @members.find { |member| member["login"] == login }
+  hsh["active"] = true
+end
+
+# for each repo
+@repos.each do |repo|
+
+  print "analyzing #{repo["full_name"]}"
+
+  # get all commits after specified date and iterate
+  print "...commits"
+  begin
+    @client.commits_since(repo["full_name"], ARGV[1]).each do |commit|
```

Comment:
👋 @pholleran Thanks for working on this! 🌟 Just wondering (since I don't know): will this find commits only in the default branch or in all branches? 💭 If it will find commits only in the default branch -- is that the behavior users will expect, related to how inactivity is defined? For example, a user might start working on a feature branch which hasn't been merged into master or they might have not opened a pull request for that branch yet. 

As far as I know, we don't have an API endpoint which returns commits from multiple branches -- you can only fetch commits from a single specific branch (https://developer.github.com/v3/repos/commits/#list-commits-on-a-repository). So, I'm _guessing_ this method will return commits from the default branch only, or that it iterates over all branches in the repository, then combines commits from all those branches on the client side, and then returns those.

Reply:
@izuzak Just the default branch. We should document this.